examples/tensorflow-profiling-examples/trainer-deep-profiling/model.py

```diff
@@ -0,0 +1,151 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+import tensorflow as tf
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+BUCKET = None  # set from task.py
+PATTERN = 'of'  # gets all files
+TRAIN_STEPS = 10000
+CSV_COLUMNS = 'weight_pounds,is_male,mother_age,plurality,gestation_weeks,key'.split(',')  # noqa: E501
+LABEL_COLUMN = 'weight_pounds'
+KEY_COLUMN = 'key'
+DEFAULTS = [[0.0], ['null'], [0.0], ['null'], [0.0], ['nokey']]
+
+# Define some hyperparameters
+BATCH_SIZE = 512
+N_EMBEDS = 3
+FIRST_LAYER_SIZE = 64
+NUM_LAYERS = 3
+DROPOUT_RATE = 0.1
+LEARNING_RATE = 0.001
+EVAL_INTERVAL = 200
+
+TRANSFORMED_TRAIN_DATA_FILEBASE = 'train_transformed'
+TRANSFORMED_TEST_DATA_FILEBASE = 'test_transformed'
+EXPORTED_MODEL_DIR = 'exported_model_dir'
+
+
+def read_dataset(mode):
+    def _decode_csv(line):
+        features = tf.decode_csv(line, record_defaults=DEFAULTS)
+        features = dict(zip(CSV_COLUMNS, features))
+        labels = features.pop(LABEL_COLUMN)
+        return features, labels
+
+    def _input_fn():
+        # use prefix to create filename
+        if mode == tf.estimator.ModeKeys.TRAIN:
+            prefix = 'train'
+            num_epochs = None
+            shuffle = True
+        else:
+            prefix = 'eval'
+            num_epochs = 1
+            shuffle = False
+        filename_pattern = 'gs://{}/preproc/{}*{}*'.format(BUCKET, prefix, PATTERN)
+        files = tf.data.Dataset.list_files(file_pattern=filename_pattern,
+                                           shuffle=shuffle)
+        dataset = files.flat_map(lambda filename: tf.data.TextLineDataset(filename).map(_decode_csv))
+        if mode == tf.estimator.ModeKeys.TRAIN:
+            dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)
+        else:
+            dataset = dataset.batch(5012)
+        return dataset.repeat(num_epochs).make_one_shot_iterator().get_next()
+
+    return _input_fn
+
+
+def get_wide_deep(n_embeds):
+    # define column types
+    plurality_values = ['Single(1)', 'Twins(2)', 'Triplets(3)',
+                        'Quadruplets(4)', 'Quintuplets(5)', 'Multiple(2+)']
+    is_male = tf.feature_column.categorical_column_with_vocabulary_list('is_male', ['True', 'False', 'Unknown'])
+    mother_age = tf.feature_column.numeric_column('mother_age')
+    plurality = tf.feature_column.categorical_column_with_vocabulary_list('plurality', plurality_values)
+    gestation_weeks = tf.feature_column.numeric_column('gestation_weeks')
+
+    # discretize
+    age_buckets = tf.feature_column.bucketized_column(mother_age,
+                                                      boundaries=np.arange(15, 45, 1).tolist())  # pylint: disable=C0301
+    gestation_buckets = tf.feature_column.bucketized_column(gestation_weeks,
```

Comment:
The pylint disable should not be required:

    gestation_buckets = tf.feature_column.bucketized_column(
        gestation_weeks, boundaries=np.arange(17, 47, 1).tolist())

Reply:
ack